pkg/controller/persistentvolume/persistentvolume_claim_binder_controller.go

```diff
@@ -323,11 +326,15 @@ func syncVolume(volumeIndex *persistentVolumeOrderedIndex, binderClient binderCl
 	case api.VolumeReleased:
 		if volume.Spec.ClaimRef == nil {
 			return fmt.Errorf("PersistentVolume[%s] expected to be bound but found nil claimRef: %+v", volume.Name, volume)
-		} else {
-			// another process is watching for released volumes.
-			// PersistentVolumeReclaimPolicy is set per PersistentVolume
-			//  Recycle - sets the PV to Pending and back under this controller's management
-			//  Delete - delete events are handled by this controller's watch. PVs are removed from the index.
+		}
+		// another process is watching for released volumes.
+		// PersistentVolumeReclaimPolicy is set per PersistentVolume
+		//  Recycle - sets the PV to Pending and back under this controller's management
+		//  Delete - delete events are handled by this controller's watch. PVs are removed from the index.
+		_, err := binderClient.GetPersistentVolumeClaim(volume.Spec.ClaimRef.Namespace, volume.Spec.ClaimRef.Name)
```

Comment:
Volumes in `Retain` phase must stay retain forever and never get bound again. See the example above - if I create new pvc (and pod), I must not see any old data there.